extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticationMechanism.java

```diff
@@ -20,11 +20,10 @@ public interface HttpAuthenticationMechanism {
 
     Uni<ChallengeData> getChallenge(RoutingContext context);
 
-    /**
-     * Returns the required credential types. If there are no identity managers installed that support the
-     * listed types then this mechanism will not be enabled.
-     */
-    Set<Class<? extends AuthenticationRequest>> getCredentialTypes();
+    @Deprecated
```

Comment:
Not sure we should deprecate as it still can help IdentityProviderManager

Reply:
Alright, I didn't know about such plans. All I saw was completely unused method. I'll drop the deprecation.